apps/mesh/src/storage/connection.ts

```
@@ -56,10 +56,23 @@ export class ConnectionStorage implements ConnectionStoragePort {
     private vault: CredentialVault,
   ) {}
 
+  private isValidConnectionId(id?: string): boolean {
+    if (!id) return false;
+    return /^conn_[a-zA-Z0-9_-]+$/.test(id);
+  }
+
   async create(data: Partial<ConnectionEntity>): Promise<ConnectionEntity> {
-    const id = `conn_${nanoid()}`;
+    const id = this.isValidConnectionId(data.id)
```

you should throw an error in case of invalid id.